src/Tribe/Values/Abstract_Value.php

```diff
@@ -0,0 +1,455 @@
+			 *
+			 * @param float $value the normalized value
+			 * @param Abstract_Value the object instance
+			 *
+			 * @return float
+			 */
+			$value = (float) apply_filters( "tec_common_value_{$this->class_name}_normalized", (float) $value, $this );
+
+			/**
+			 * Filter the value returned for normalize() when implemented in a specific class name
+			 *
+			 * @since TBD
+			 *
+			 * @param float $value the normalized value
+			 * @param Abstract_Value the object instance
+			 *
+			 * @return float
+			 */
+			do_action( 'tec_common_value_normalize_is_numeric', $value, $this );
+
+			/**
+			 * Filter the value returned for get_precision() when implemented in any class
+			 *
+			 * @since TBD
+			 *
+			 * @param float $value the normalized value
+			 * @param Abstract_Value the object instance
+			 *
+			 * @return float
+			 */
+			return (float) apply_filters( 'tec_common_value_normalized', $value, $this );
+		}
+
+		if ( $this->is_character_block( $value ) ) {
+			return (float) 0;
+		}
+
+		// If we can split the amount by spaces, remove any blocks that don't contain any digits
+		// This is important in case the currency unit contains the same characters as the decimal/thousands
+		// separators such as in Moroccan Dirham (1,234.56 .د.م.) or Danish Krone (kr. 1.234,56)
+		foreach ( explode( ' ', $value ) as $block ) {
+			if ( $this->is_character_block( $block ) ) {
+				$value = str_replace( $block, '', $value );
+			}
+		}
+
+		// Remove encoded html entities
+		$value = preg_replace( '/&[^;]+;/', '', trim( $value ) );
+
+		// Get all non-digits from the amount
+		preg_match_all( '/[^\d]/', $value, $matches );
+
+		// if the string is all digits, it is numeric
+		if ( empty( $matches[0] ) ) {
+			return (float) $value;
+		}
+
+		$tokens    = array_unique( $matches[0] );
+		$separator = '/////';
+
+		foreach ( $tokens as $token ) {
+			if ( $this->is_decimal_separator( $token, $value ) ) {
+				$separator = $token;
+				continue;
+			}
+
+			$value = str_replace( $token, '', $value );
+		}
+
+		$pieces = explode( $separator, $value );
+
+		// If the initial amount did not have decimals specified, $pieces will be an array of a single
+		// numeric value, so we just return it as a float.
+		if ( 1 === count( $pieces ) && is_numeric( reset( $pieces ) ) ) {
+			return (float) reset( $pieces );
+		}
+
+		$decimal = array_pop( $pieces );
+
+		return (float) implode( '', array_merge( $pieces, [ '.', $decimal ] ) );
+	}
+
+	/**
+	 * Value loader. This method uses Reflection to call all registered `set_$property_value` methods in the
+	 * inheritance chain every time the object is updated, so the values in each of the formats are always kept up to
+	 * date.
+	 *
+	 * @since TBD
+	 */
+	private function update() {
+		$reflection = $this->get_reflection_class( $this );
+
+		foreach ( $this->get_setters() as $setter ) {
+			$method = $reflection->getMethod( $setter );
+			$method->setAccessible( true );
+			$method->invoke( $this );
+		}
+	}
```

Comment:
Same concept, take the setters from the perspective of a list of callable, and if they are not accessible it just doesn't work. No need to force the accessibility. I know it changes how certain methods are used, but it prevents the reflection, which can add complexity to the code flow for less experienced developers.